endpoint.go

```diff
@@ -316,7 +311,7 @@ func (ep *endpoint) hasInterface(iName string) bool {
 	ep.Lock()
 	defer ep.Unlock()
 
-	for _, iface := range ep.iFaces {
+	if iface := ep.iface; iface != nil {
```

Comment:
Can we refactor this to the following one line: 
`return ep.iface != nil && ep.iface.srcName == iName`

No more need for a local iFace variable and  if block.